plugins/inputs/nats_top/nats_top.go

```diff
@@ -0,0 +1,93 @@
+package natstop
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+
+	"encoding/json"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	gnatsd "github.com/nats-io/gnatsd/server"
+)
+
+type NatsTop struct {
+	Server string
+}
+
+var sampleConfig = `
+  ## The address of the monitoring end-point of the NATS server
+  server = "http://localhost:1337"
+`
+
+func (n *NatsTop) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *NatsTop) Description() string {
+	return "Provides metrics about the state of a NATS server"
+}
+
+func (n *NatsTop) Start(acc telegraf.Accumulator) error {
```

Comment:
Since there is nothing to do in the Start and Stop functions you should probably not implement them.

Reply:
Got it.